receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -319,3 +334,60 @@ func (mf *metricFamily) appendMetric(metrics pmetric.MetricSlice) {
 
 	metric.MoveTo(metrics.AppendEmpty())
 }
+
+func (mf *metricFamily) addExemplar(l labels.Labels, e exemplar.Exemplar) error {
+	gk := mf.getGroupKey(l)
+	mg := mf.groups[gk]
+	if mg == nil {
+		return nil
+	}
+	es := mg.exemplars
+	err := convertToOtlpExemplar(e, es.AppendEmpty())
+	return err
+}
+
+func convertToOtlpExemplar(pe exemplar.Exemplar, e pmetric.Exemplar) error {
+	e.SetTimestamp(timestampFromMs(pe.Ts))
+	e.SetDoubleVal(pe.Value)
+	e.FilteredAttributes().EnsureCapacity(len(pe.Labels))
+	for _, lb := range pe.Labels {
+		switch strings.ToLower(lb.Name) {
+		case "trace_id", "traceid":
+			var tid [16]byte
+			b, err := hex.DecodeString(lb.Value)
```

Comment:
Consider to use https://pkg.go.dev/encoding/hex#Decode to avoid a temporary string/buffer.

Reply:
Not sure because we need a temporary variable because we need to copy to lower bytes in case of length less than required. Please let me know if you think otherwise

Reply:
I think otherwise :). Something like this will work

```golang
srcLen := DecodeLen(src)
if len(dst) >= srcLen {
  hex.Decode(dst[len(dst) - srcLen:], src)
} else {
    hex.Decode(dst, src[:hex.EncodeLen(dst)])
}
```

Reply:
Done, implemented